src/curve/stable/StablePair.sol

```diff
@@ -109,7 +109,10 @@ contract StablePair is ReservoirPair {
     function mint(address) external override returns (uint256) {
         // DELEGATE TO StableMintBurn
         address lTarget = MINT_BURN_LOGIC;
-        assembly {
+
+        // SAFETY:
+        // This assembly block is memory safe as it does not return to solidity
```

Comment:
Apparently, the solidity docs claim that even if an assembly block does not return to solidity it is not memory safe... So we might need to leave these annotations off which baffles me...

> Since this is mainly about the optimizer, these restrictions still need to be followed, even if the assembly block reverts or terminates. As an example, the following assembly snippet is not memory safe, because the value of returndatasize() may exceed the 64 byte scratch space:

https://docs.soliditylang.org/en/v0.8.16/assembly.html

Reply:
Okay so here's my life of logic. Correct me if my line of thought is wrong. 

What you quoted (solidity docs) serves as an example generally. HOWEVER, in our case in `StableMintBurn`, both calldata and returndata fit within 64 bytes as per my explanation [here](https://github.com/vexchange/v3-core/pull/150#discussion_r1081023287). 

So the reason that it is memory safe is that they both don't exceed 64 bytes, not that it doesn't release back to solidity. UNLESS, `returndatasize` does not respect `(uint256, uint256)` as specified in `StableMintBurn::burn`. We don't expect this to happen do we? 

^ is my line of logic above sensible?

Reply:
That's true if lTarget == MINT_BURN_LOGIC which seems fairly likely we can control. I would write a comment to that effect

Reply:
okay, have updated the comment